online-shop/src/app/cart/cart.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { ActivatedRoute } from '@angular/router';
+import { Post } from '../product/post';
+import { CartService } from '../cart.service';
+
+@Component({
+  selector: 'app-cart',
+  templateUrl: './cart.component.html',
+  styleUrls: ['./cart.component.css'],
+})
+export class CartComponent implements OnInit {
+  cart = new Array<Post>();
+
+  constructor(
+    private route: ActivatedRoute,
+    private cartService: CartService,
+    private http: HttpClient
+  ) {}
+  ngOnInit(): void {
+    this.cart = this.cartService.getCart();
+  }
+
+  checkout(): void {
+    window.alert('Comanda efectuata!');
+
+    this.cartService.checkout().subscribe(() => {});
```

Comment:
**suggestion:** take care to avoid memory leaks in your code by unsubscribing from observables. https://www.thisdot.co/blog/best-practices-for-managing-rxjs-subscriptions/